### PR TITLE
Add Azure Functions templates to collection-index.yml

### PR DESCRIPTION
**Pull request by shibayan**
I have created a new modern template to replace the obsolete Azure Functions template.

### CODE REVIEW

File: _data/collection-index.yml [hunk 1]
```diff
@@ -638,3 +638,8 @@
   contact: https://github.com/lumenpink/devcontainer-features/issues
   repository: https://github.com/lumenpink/devcontainer-features
   ociReference: ghcr.io/lumenpink/devcontainer-features
+- name: Azure Functions Dev Container Templates
+  maintainer: Tatsuro Shibamura
+  contact: https://github.com/shibayan/devcontainers/issues
+  repository: https://github.com/shibayan/devcontainers
```
Comment (eljog) [inline _data/collection-index.yml:659]: Please consider adding a readme.
https://github.com/devcontainers/feature-starter has a good reference

**Review by eljog — COMMENTED** [review]
Thank you for your contribution.
I have added a comment

File: _data/collection-index.yml (hunk 1 above)
Comment (shibayan) [reply _data/collection-index.yml:659]: Thank you for your review comment. I have added a README that clearly explains the template content.

**Comment by shibayan** [comment]
@eljog Thanks! I have corrected the points you pointed out and conflicts.

**Review by eljog — DISMISSED** [review]

**Comment by shibayan** [comment]
Fixed conflicts

**Review by samruddhikhandale — APPROVED** [review]
Looks great, thanks for the addition! 🚀

**Pull request merged**